azerothcore/azerothcore-wotlk
# fix(CORE/Player): Book of the Dead unequip fix

## Changes Proposed:
-  Adds the Summon Skeleton effect of Book of the Dead to unequip exceptions list
-  Allows player to unequip Book of the Dead and have the summoned skeleton remain for its duration

## Issues Addressed:

- Closes https://github.com/azerothcore/azerothcore-wotlk/issues/11691
- Closes https://github.com/chromiecraft/chromiecraft/issues/3480
- Also fixes Ancient Cornerstone Grimoire

## SOURCE:

Additional source material in the above issue links.
https://www.wowhead.com/item=13353/book-of-the-dead#comments:id=239392

## Tests Performed:

- Used active ability of Book of the Dead, swapped to a different offhand equip, confirmed summoned skeleton remained active.


## How to Test the Changes:


1. Create/Select character level 58+
2. .additem 17067
3. Equip Book of the Dead, wait until active ability is ready
4. Use active ability
5. Swap to any other offhand item or otherwise unequip Book of  the Dead
6. Observe summoned skeleton remains active
7. Observe summoned skeleton correctly disappears after its full 1minute duration

## Known Issues and TODO List:




## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

**REMEMBER**: when testing a PR that changes something **generic** (i.e. a part of code that handles more than one specific thing), the tester should not only check that the PR does its job (e.g. fixing spell XXX) but **especially** check that the PR does not cause any regression (i.e. introducing new bugs).

**For example**: if a PR fixes spell X by changing a part of code that handles spells X, Y, and Z, we should not only test X, but **we should test Y and Z as well**.
